lib/location/method_cellular.c

```diff
@@ -44,17 +44,21 @@ void method_cellular_lte_ind_handler(const struct lte_lc_evt *const evt)
 	switch (evt->type) {
 	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS: {
 		LOG_DBG("Cell measurements results received");
-		memset(&cell_data, 0, sizeof(struct lte_lc_cells_info));
 
-		if (evt->cells_info.current_cell.id) {
-			/* Copy current and neighbor cell information */
-			memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
-			memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
-			       sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+		/* Copy current cell information. */
+		memcpy(&cell_data.current_cell,
+		       &evt->cells_info.current_cell,
+		       sizeof(struct lte_lc_cell));
+
+		/* Copy neighbor cell information if present. */
+		if (evt->cells_info.ncells_count > 0 && evt->cells_info.neighbor_cells) {
+			memcpy(neighbor_cells,
```

Comment:
I think it would be nicer if `neightbor_cells` would be accessed through the pointer in `cell_data`:
```suggestion
			memcpy(cell_data.neighbor_cells,
```

Reply:
Agree. Changed.